ledger/acctupdates.go

```diff
@@ -1971,6 +1974,61 @@ func (au *accountUpdates) commitRound(offset uint64, dbRound basics.Round, lookb
 
 }
 
+// compactDeltas takes an arary of account map deltas ( one array entry per round ), and corresponding creatables array, and compact the arrays into a single
+// map that contains all the account deltas changes. While doing that, the function eliminate any intermediate account changes.
+// The function returns a map that could be the same as the input map, and therefore both the outputs of this function as well as the inputs to this function
+// should be kept read-only ( unless we don't care about the correctness of the data anymore ).
+func compactDeltas(accountDeltas []map[basics.Address]accountDelta, creatableDeltas []map[basics.CreatableIndex]modifiedCreatable) (outAccountDeltas map[basics.Address]accountDelta, outCreatableDeltas map[basics.CreatableIndex]modifiedCreatable) {
+	// this is a common case where we have only a single round that we try to flush. In this case, we're just going to return
+	// the first entry.
+	if len(accountDeltas) == 1 && len(creatableDeltas) == 1 {
```

Comment:
the length of both arrays is always identical ( look at the caller ), and should always be 1 or greater.
For functional correctness, what you've suggested won't be correct since we're indexing the first entry inside this if clause, which would cause an access violation. I think that the `len=0` is already handled correctly in the subsequent two if statements.